kobo/admin/templates/client/__project_name__.conf

```diff
@@ -3,7 +3,7 @@
 # Hub XML-RPC address.
 HUB_URL = "http://localhost:8000/xmlrpc"
 
-# Hub authentication method. Example: krbv, password, worker_key
+# Hub authentication method. Example: krbv, gssapi, password, worker_key
```

Comment:
Does gssapi really need to be exposed as a new method?  A few points to make about this:

1. Config files for clients with AUTH_METHOD="krbV" will already be widely deployed.  So if the intent here is to eventually phase out the krbV library and migrate to gssapi, we would have to somehow arrange for every client to update their config file before we could finally drop krbV from the server.

2. It's currently requiring both the client and the server to use the same library (either krbV or gssapi).  But it's not clear to me if that's needed at all.  The reason it's not clear is that the login_krbv, login_gssapi XML-RPC methods don't document their arguments.  Question: if the client's krbV code path called login_gssapi on hub, or the client's gssapi code path called login_krbv on the hub, does that work?  If so, what is the logic in artificially restricting it so that the client and server need to use the same library, given that those may not even be the same OS or major Python version.

3. I'm not sure if there is a need to continue supporting multiple libraries at all.  gssapi already seems to support all versions of Python supported by kobo (note kobo master doesn't support py2.6).

How about just dropping krbV, and letting the "krbv" login method use gssapi instead?  krbv can be interpreted as referring to "kerberos 5" and not specifically the Python library named krbV.

Reply:
My original implementation has completely replaced krbV with gssapi, but suggestion above ^^ has made me introduce it as a new method. As I have mentioned in a previous comment, I don't have a good idea on how to test this change, and I cannot guarantee that it performs the exact same actions as krbV method. The API is quite different between the two packages, so it was difficult for me to ensure that the performed actions are 1:1. So right now, the new library opt-in approach seems reasonable to me. As for your point 2.), I doubt that this could work. The inputs and outputs are different.

I'm not an expert on the ins and outs of Kerberos authentication, but if we can ensure that it always works as expected, then I am okay with either approach.

Reply:
In response to @rohanpm:
1) Yes, that is true.
2) `krbV` and `gssAPI` does not work together (unless you use some nasty hacks). So if the `krbV` code path on client calls `login_gssapi`  on hub (or the other way), it don't work. In my opinion it is *not* a good idea to presserve the name of the authentication method (`auth_krbV`) when reimplemented by `gssApi`. If those implementations don't work together, I think it is not reasonable to call those authentication methods the same.
3) Technically it is not needed, but it would make transition from `krbV` to `gssAPI` much smoother (at least in my opinion). Some clients using older versions of `kobo` simply don't have implementation for `gssAPI`. Yes, they could be upgraded, but it would have to happen at the same time as update on the server side. Otherwise they would stop working at the moment of kobo upgrade on server.